src/main/java/gregtech/api/unification/material/info/MaterialFlags.java

```diff
@@ -0,0 +1,225 @@
+package gregtech.api.unification.material.info;
+
+import gregtech.api.unification.material.Material;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class MaterialFlags {
+
+    private final Set<MaterialFlag> flags = new HashSet<>();
+
+    public MaterialFlags addFlags(MaterialFlag... flags) {
+        this.flags.addAll(Arrays.asList(flags));
+        return this;
+    }
+
+    public void verify(Material material) {
+        flags.addAll(flags.stream()
+                .map(f -> f.verifyFlag(material))
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet()));
+    }
+
+    public boolean hasFlag(MaterialFlag flag) {
+        return flags.contains(flag);
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        flags.forEach(f -> sb.append(f.toString()).append("\n"));
+        return sb.toString();
+    }
+
+    /////////////////
+    //   GENERIC   //
+    /////////////////
+
+    /**
+     * Add to material to disable it's unification fully
+     * todo implement
+     */
+    public static final MaterialFlag NO_UNIFICATION =
+            new MaterialFlag.Builder(0, "no_unification").build();
+
+    /**
+     * Decomposition recipe requires hydrogen as additional input. Amount is equal to input amount
+     */
+    public static final MaterialFlag DECOMPOSITION_REQUIRES_HYDROGEN =
+            new MaterialFlag.Builder(1, "decomposition_requires_hydrogen").build();
+
+    /**
+     * Enables electrolyzer decomposition recipe generation
+     */
+    public static final MaterialFlag DECOMPOSITION_BY_ELECTROLYZING =
+            new MaterialFlag.Builder(2, "decomposition_by_electrolyzing").build();
+
+    /**
+     * Enables centrifuge decomposition recipe generation
+     */
+    public static final MaterialFlag DECOMPOSITION_BY_CENTRIFUGING =
+            new MaterialFlag.Builder(3, "decomposition_by_centrifuging").build();
+
+    /**
+     * Disables decomposition recipe generation for this material and all materials that has it as component
+     */
+    public static final MaterialFlag DISABLE_DECOMPOSITION =
+            new MaterialFlag.Builder(4, "disable_decomposition").build();
+
+    /**
+     * Add to material if it is some kind of explosive
+     */
+    public static final MaterialFlag EXPLOSIVE =
+            new MaterialFlag.Builder(5, "explosive").build();
+
```

Comment:
Why not move the `BURNING` flag? It would allow for damage ticks when carrying hot fluid cells/buckets and also would be useful for hot ingots.

Reply:
It was not being used for anything, so I just removed it. I think adding functionality to more flags would be good, but I just trimmed out what we had before